CHANGELOG.md

```diff
@@ -2,6 +2,14 @@
 
 #### Upcoming Changes
 
+* Refactor `Refactor verify_secure_runner` [#768](https://github.com/lambdaclass/cairo-rs/pull/768)
+    Public Api changes:
+    * Remove builtin name from the return value of `BuiltinRunner::get_memory_segment_addresses`
+    * Simplify the return value of `CairoRunner::get_builtin_segments_info` to `HashMap<isize, usize>`
+    * CairoRunner::read_return_values now receives a mutable reference to VirtualMachine
+    Bugfixes:
+    * CairoRunner::read_return_values now updates the `stop_ptr` of each builtin after calling `BuiltinRunner::final_stack`
```

Comment:
This could be a bit confusing. When I call a method called `read_something` I don't expect it to change stage.

Reply:
The reason for this is the following:
`BuiltinRunner::final_stack`, in the source code updates the value of the builtin's stop_ptr. When this method was implemented, they needed the method to receive an inmutable reference (i dont know the reason why). So they instead made the method return the stop_ptr as a second return value and added the method `BuiltinRunner::stop_ptr` to update the stop_ptr after calling `BuiltinRunner::final_stack`. But this didnt happen, leading to the stop_ptr never being written and therefore verify_secure_runner always returned with error due to stop_ptr being None.
The correct way to fix this would be to correct `BuiltinRunner::final_stack`, but this requires changes in cairo_rs_py too, thats why I did a quick fix for the moment.

Reply:
Can we point the quick fix with a `FIXME` comment?

Reply:
sure